cmake/Platform/Common/GCC/Configurations_gcc.cmake

```diff
@@ -73,6 +74,15 @@ ly_append_configurations_options(
         -Wno-switch
         -Wno-uninitialized
         -Wno-unused-result
+        -Wno-dangling-pointer
```

Comment:
This shouldnt be needed with that warning disable statement above

Reply:
```
In file included from /media/slapin/data/o3de-projects/NewspaperDeliveryGame/build/linux/o3de/Code/Framework/AzCore/CMakeFiles/AzCore.dir/Unity/unity_15_cxx.cxx:15:
/media/slapin/data/o3de/Code/Framework/AzCore/AzCore/Metrics/EventLoggerReflectUtils.cpp: In function ‘AZ::Metrics::Script::ReflectRecordEvent(AZ::BehaviorContext&)::<lambda(AZ::Metrics::EventLoggerId, AZ::Metrics::EventPhaseNamespace::EventPhase, AZ::Metrics::Script::ScriptEventArgs, AZ::Metrics::IEventLoggerFactory*)>’:
/media/slapin/data/o3de/Code/Framework/AzCore/AzCore/Metrics/EventLoggerReflectUtils.cpp:465:66: note: unnamed temporary defined here
  465 |                     asyncArgs.m_id = eventPhaseArgs.m_id.value_or(AZStd::string_view{});
      |                                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
```

No, that is still needed